orchestra/machine_tasks.py

```diff
@@ -85,7 +92,13 @@ def execute(project_id, step_slug):
             {'data': task_assignment.in_progress_task_data,
              'datetime': timezone.now().isoformat(),
              'type': TaskAssignment.SnapshotType.SUBMIT,
-             'work_time_seconds': 0
-             })
+             'work_time_seconds': 0})
         task_assignment.save()
+
+        iteration = ordered_iterations(task).last()
+        iteration.status = Iteration.Status.REQUESTED_REVIEW
+        iteration.submitted_data = task_data
+        iteration.end_datetime = timezone.now()
+        iteration.save()
```

Comment:
If task isn't aborted, save iteration data after machine (or multiple machines, if retried) executes task function

(Not sure whether the iteration should be completed if task is aborted, open to suggestions)